packages/firestore/test/unit/specs/recovery_spec.test.ts

```diff
@@ -62,24 +62,21 @@ describeSpec('Persistence Recovery', ['no-ios', 'no-android'], () => {
     ['multi-client'],
     () => {
       const query = Query.atPath(path('collection'));
+      const doc1 = doc('collection/doc', 1, { foo: 'a' });
 
       return client(0)
         .expectPrimaryState(true)
         .client(1)
         .expectPrimaryState(false)
         .userListens(query)
-        .failDatabaseTransactions({
-          'Allocate target': true,
-          'Lookup mutation documents': true,
-          'Get new document changes': true
-        })
+        .failDatabaseTransactions('Get new document changes')
```

Comment:
Same concern here: previously this failed to allocate a target at all, which should have caused the listen to fail. Now this is only failing much later.

Reply:
Similar response: The target is already allocated and this seems like a copy/paste issue. Failed target allocations are tested in 'Fails targets that cannot be allocated'. 
I am not 100% convinced we need multi-tab specific tests for these initial failures, since these failures happen before we kick off any multi-tab specific failures. If we add a test, it would be a sanity check to ensure that we  don't write a Local Storage notification if we failed to persist the target/write.

Reply:
OK. I see it now. Thanks for talking it through with me.